app/models/connected_apps/vine.rb

```diff
@@ -4,12 +4,18 @@
 #
 module ConnectedApps
   class Vine < ConnectedApp
-    def connect(api_key: )
-      # TODO
-    end
+    VINE_API_URL = "https://vine-staging.openfoodnetwork.org.au/api/v1/my-team"
+
+    def connect(api_key:, vine_api:, **_opts)
+      response = vine_api.my_team
+
+      return update data: { api_key: } if response.success?
 
-    def disconnect
-      # TODO
+      errors.add(:base, I18n.t("activerecord.errors.models.connected_apps.vine.api_request_error"))
```

Comment:
Can you not just add the error name and let it get translated automatically?
```suggestion
      errors.add(:base, :api_request_error)
```
The resulting key might be slightly different.

Reply:
I tried that, I think , but I couldn't get it to work. I'll have another go

Reply:
according to this  : https://guides.rubyonrails.org/i18n.html#error-message-scopes it should work but for some reason it doesn't ?? It's looking for key "en_AU.api_request_error"
I am leaving as it is.

Reply:
🤦 Sorry don't know what I was thinking, that probably should have been using lazy-lookup:

```suggestion
      errors.add(:base, ".api_request_error")
```

Anyway, no big deal  if it doesn't work.